### PR TITLE
Implement thumbnail interfaces

### PR DESCRIPTION
**Pull request by kjkooistra-youwe**
Added interfaces for thumbnails, since the classes are final.
This allows using the interfaces for proxies, like unit test mock objects.
Relates to #13595

### CODE REVIEW

**Comment by kjkooistra-youwe** [comment]
This is an initial approach. I haven't been able to test it yet.

**Comment by kjkooistra-youwe** [comment]
Tested image and video thumbnails in the latest 11.x version. Seems to work.

File: models/Asset/Thumbnail/ImageThumbnailInterface.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    public function getWidth(): int;
+
+    public function getHeight(): int;
+
+    public function getRealWidth(): int;
+
+    public function getRealHeight(): int;
+
+    public function getDimensions(): array;
```
Comment (kingjia90) [inline models/Asset/Thumbnail/ImageThumbnailInterface.php:45]: Would it make sense to refactor this `ImageThumbnailTrait` into an abstract class or so? 

https://github.com/pimcore/pimcore/blob/11.x/models/Asset/Thumbnail/ImageThumbnailTrait.php#L190

File: models/Asset/Document/ImageThumbnailInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\Asset\Document;
+
+use Pimcore\Model\Asset\Thumbnail\ThumbnailInterface;
+use Pimcore\Model\Asset\Thumbnail\ThumbnailPathInterface;
+
+interface ImageThumbnailInterface extends ThumbnailInterface, ThumbnailPathInterface
```
Comment (kingjia90) [inline models/Asset/Document/ImageThumbnailInterface.php:23]: Could we find some different names, for example, currently is
`ImageThumbnailInterface` extends `ThumbnailInterface` which extends `ImageThumbnailInterface`.

The first and third interface have same name but different namespaces but could be confusing.

see also https://github.com/pimcore/pimcore/pull/14337/files?w=1#diff-9b0c5f8a3ce734cadaa7f15f3afe14ab9c7d51ec84cdf4f9bf45b21e830a8f47R20
and here where an alias would be required
https://github.com/pimcore/pimcore/pull/14337/files?w=1#diff-09eb7bedc63f66656f8b3dfa9c256dad2cb608b933b26a797703cf6bbdcf48a1R20

**Review by kingjia90 — COMMENTED** [review]
So far LGTM, tests are all green as well, just some inputs while we are at it

**Review by dvesh3 — APPROVED** [review]
Looks good, other than changes requested by @kingjia90

File: models/Asset/Thumbnail/ImageThumbnailInterface.php (hunk 1 above)
Comment (kingjia90) [reply models/Asset/Thumbnail/ImageThumbnailInterface.php:45]: postponing these suggestions

File: models/Asset/Document/ImageThumbnailInterface.php (hunk 2 above)
Comment (kingjia90) [reply models/Asset/Document/ImageThumbnailInterface.php:23]: postponing these suggestions

**Pull request merged**